path_config.go

```diff
@@ -0,0 +1,232 @@
+// Copyright © 2019, Oracle and/or its affiliates.
+package ociauth
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+	"strings"
+)
+
+//These constants store the configuration keys
+const (
+	HomeTenancyIdConfigName = "homeTenancyId"
+)
+
+func pathConfig(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "config",
+		Fields: map[string]*framework.FieldSchema{
+			HomeTenancyIdConfigName: {
+				Type:        framework.TypeString,
+				Description: "The tenancy id of the account.",
+			},
+		},
+
+		ExistenceCheck: b.pathConfigExistenceCheck,
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.CreateOperation: b.pathConfigCreate,
+			logical.UpdateOperation: b.pathConfigUpdate,
+			logical.DeleteOperation: b.pathConfigDelete,
+			logical.ReadOperation:   b.pathConfigRead,
+		},
+
+		HelpSynopsis:    pathConfigSyn,
+		HelpDescription: pathConfigDesc,
+	}
+}
+
+// Establishes dichotomy of request operation between CreateOperation and UpdateOperation.
+// Returning 'true' forces an UpdateOperation, CreateOperation otherwise.
+func (b *backend) pathConfigExistenceCheck(ctx context.Context, req *logical.Request, data *framework.FieldData) (bool, error) {
+	entry, err := b.lockedOCIConfig(ctx, req.Storage)
+	if err != nil {
+		return false, err
+	}
+	return entry != nil, nil
+}
+
+// lockedOCIConfig returns the properties set on the given config. This method
+// acquires the read lock before reading the config from the storage.
+func (b *backend) lockedOCIConfig(ctx context.Context, s logical.Storage) (*OCIConfigEntry, error) {
```

Comment:
The "locked" versions can be removed, per previous discussion.

Reply:
Done